rs/replicated_state/src/canister_state.rs

```diff
@@ -369,12 +369,14 @@ impl CanisterState {
 
     /// The amount of memory currently being used by the canister.
     ///
-    /// This only includes execution memory (heap, stable, globals, Wasm),
-    /// canister history memory and wasm chunk storage.
+    /// This includes execution memory (heap, stable, globals, Wasm),
+    /// canister history memory, wasm chunk storage and snapshots that
+    /// belong to this canister.
     pub fn memory_usage(&self) -> NumBytes {
```

Comment:
I just needed to disentangle in my head difference between `memory_usage` and `snapshot_memory_usage`, also in this file. While it makes sense in the end that they are different, could we improve the comments a bit to point out specifically how they are different and relate to each other?

Reply:
I'm not sure I'm following what you mean. `snapshot_memory_usage` is one more thing that's included in the canister's total memory usage, like heap, stable, etc.

Reply:
I think you're proving my point. There is `memory_usage(&self)`, `snapshot_memory_usage(&self)`, and `system_state.snapshots_memory_usage`. 

`memory_usage(&self)` includes `system_state.snapshots_memory_usage`, but `snapshot_memory_usage(&self)` is similar to `memory_usage(&self)` but does not include `system_state.snapshots_memory_usage`.

Reply:
Oh I see, now I understand what you mean. `snapshot_memory_usage(&self)` returns the memory usage of the newly created snapshot based on the current canister's state (if the canister state evolves, this function will return something new). `system_state.snapshots_memory_usage` is the cumulative usage of all snapshots associated with this canister .

Reply:
I gave it a shot and adjusted doc comments to include more information.

Reply:
I also find it potentially confusing that we have `memory_usage()`, `execution_memory_usage()`, `message_memory_usage()`, etc., all with the same semantics; and then `snapshot_memory_usage()`, meaning a completely different thing.

In addition to the doc comment, I would also change the name to something like `snapshot_size_bytes()`, or anything other than `<X>_memory_usage()`.